[generate_opcheck_tests] tests should ignore meta/FakeTensors

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109641
* #109640
* #109639
* #109638
* #109637

These tests generally don't work on meta tensors because they need to
compare the data of the Tensors. For example, SchemaCheckMode errors out
if any inputs are meta or Fake because it needs to check their storages
to see if any mutation occurred and those do not have storages.

Test Plan:
- new tests

zou3519 committed:
[generate_opcheck_tests] test_schema should ignore meta and Fake inputs

SchemaCheckMode errors out if any inputs are meta or Fake because it
needs to check their storages to see if any mutation occurred and those
do not have storages. The autogenerated test_schema should skip the
SchemaCheckMode check if this is the case.

Test Plan:
- new tests

[ghstack-poisoned]

zou3519 committed:
Update on "[generate_opcheck_tests] test_schema should ignore meta and Fake inputs"


SchemaCheckMode errors out if any inputs are meta or Fake because it
needs to check their storages to see if any mutation occurred and those
do not have storages. The autogenerated test_schema should skip the
SchemaCheckMode check if this is the case.

Test Plan:
- new tests

[ghstack-poisoned]

zou3519 committed:
Update on "[generate_opcheck_tests] tests should ignore meta/FakeTensors"


These tests generally don't work on meta tensors because they need to
compare the data of the Tensors. For example, SchemaCheckMode errors out
if any inputs are meta or Fake because it needs to check their storages
to see if any mutation occurred and those do not have storages.

Test Plan:
- new tests

[ghstack-poisoned]

bdhirsh commented on torch/testing/_internal/optests/generate_tests.py:
```diff
@@ -21,10 +21,20 @@
+def is_abstract(tensor: torch.Tensor) -> bool:
+    if tensor.is_meta:
+        return True
+    if isinstance(tensor, torch._subclasses.fake_tensor.FakeTensor):
```
Now that subclasses can be fakeified, you probably want `fake_tensor.is_fake`

bdhirsh reviewed (approved):
Curiosity question: how did this come up? Is it because you're directly using the example inputs from internal test suites, and some of the internal tests happen to have example inputs that are meta tensors?

zou3519 commented:
> Curiosity question: how did this come up? Is it because you're directly using the example inputs from internal test suites, and some of the internal tests happen to have example inputs that are meta tensors?

Yeah. People add tests that their meta formulas are correct by running meta tensors through operators

zou3519 committed:
Update on "[generate_opcheck_tests] tests should ignore meta/FakeTensors"


These tests generally don't work on meta tensors because they need to
compare the data of the Tensors. For example, SchemaCheckMode errors out
if any inputs are meta or Fake because it needs to check their storages
to see if any mutation occurred and those do not have storages.

Test Plan:
- new tests

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge
